src/transformers/generation/logits_process.py

```diff
@@ -960,8 +957,8 @@ def __call__(self, input_ids, scores):
                     scores[k, : self.eos_token_id] = -float("inf")
 
             # apply the `max_initial_timestamp` option
-            if input_ids.shape[1] == self.begin_index and self.max_initial_timestamp_index is not None:
-                last_allowed = self.timestamp_begin + self.max_initial_timestamp_index
+            if input_ids.shape[1] == self.begin_index and self.max_initial_timestamp_idx is not None:
```

Comment:
nit: I'm under the impression Sylvain would favor `index` over `idx` (And I agree)